rust/otap-dataflow/crates/config/src/pipeline/service/telemetry/logs.rs

```diff
@@ -18,6 +19,75 @@ pub struct LogsConfig {
     /// The list of log processors to configure.
     #[serde(default)]
     pub processors: Vec<processors::LogProcessorConfig>,
+    
+    /// Internal collection for component-level logs.
+    /// 
+    /// When enabled, component logs (otel_info!, otel_warn!, etc.) are routed through
+    /// an internal telemetry receiver in the OTAP pipeline, allowing use of built-in
+    /// batch processors, retry, and exporters (console, OTLP, etc.).
+    /// 
+    /// When disabled (default), component logs are routed to the OpenTelemetry SDK,
+    /// using the same export path as 3rd party logs from tokio-tracing-rs.
+    #[serde(default)]
+    pub internal_collection: InternalCollectionConfig,
```

Comment:
Long term, I think this approach will apply not only to logs, but also to metrics and traces. So I'm wondering whether this configuration should be promoted to a more general level.
In addition, I suggest that this field be:
* an option, so we can remove the `enabled` field from `InternalCollectionConfig`, which I think would simplify things
* renamed to something more explicit. I'm not fully satisfied with my proposal `otap_pipeline`; there is probably a better name to find